src/internal_rules.jl

```diff
@@ -684,7 +685,7 @@ function EnzymeRules.forward(
     end
 end
 
-function EnzymeRules.augmented_primal(config, func::Const{typeof(cholesky)}, RT::Type, A; kwargs...)
+function EnzymeRules.augmented_primal(config, func::Const{typeof(cholesky)}, RT::Type, A::Annotation{AT}; kwargs...) where {AT <: Array}
```

Comment:
I think this should restricted 
```suggestion
function EnzymeRules.augmented_primal(config, func::Const{typeof(cholesky)}, RT::Type, A::Annotation{<:Matrix}; kwargs...)
```
or the more general
```suggestion
function EnzymeRules.augmented_primal(config, func::Const{typeof(cholesky)}, RT::Type, A::Annotation{<:AbstractMatrix}; kwargs...)
```

Reply:
Not opposed to this, but this is the style convention of the rest of the custom rules, so will be consistent for now.